### PR TITLE
Fixes Split causing remount on responsive event

### LINKED ISSUE #460: Responsive Split causes re-render  & re-fetch of data
Hello!

I've encountered a problem using `Split`.

Wrapping a component, `<C1 />`, inside a `<Split onResponsive={this._onResponsive.bind(this);}></Split>` causes `C1` to unmount and then remount(causing me to fetch data two times in `componentWillMount`) since `_onResponsive` updates the responsive-state & if the menu should be shown or not.

Do you have any solution on how to do this without re-fetching data? I want the responsive menu, but i'd like to avoid fetching more data.

I think this is more of a general React-problem rather than Grommet but i'm wondering if anyone has encountered the issue and how you solved it? I'm thinking of fetching data when a react-router route is changed, but that's not optimal IMO.

I followed [the docs-example](https://github.com/grommet/grommet-docs/blob/master/src/DocsSplit.js) but the DocsSplit doesn't fetch data so it'll only cause a re-render.

### PR DESCRIPTION
**Pull request by linde12**
This closes #460

Signed-off-by: Oscar Linde <lindeoscar@gmail.com>

### CODE REVIEW

**Comment by linde12** [comment]
This only works if there are 2 or more elements present as children i think. If you split with one child and have priority on right this won't work. Looking into it...

**Comment by linde12** [comment]
Almost done now. Fixed a bug and doing some more testing. Will put up another patch as soon as i'm ready.

**Comment by linde12** [comment]
Would love for someone to verify and review. I think this should do it. Basically what i've done is keep rendering all the components passed to Split but wrapping them in a div with a `display: none` style if they should be hidden. This prevents remounting of the component and fixes #460

**Comment by alansouzati** [comment]
hey @linde12 thanks for your contribution. let me brainstorm with this with @ericsoderberghp.

I think Tabs components have the same issue.

File: src/js/components/Split.js [hunk 1]
```diff
@@ -92,20 +92,32 @@ export default class Split extends Component {
...
+              // If priority = right and we're not
+              // the last child, then hide
+              if (idx !== children.length - 1) {
+                hidden = true;
+              }
+            }
+          }
+
+          if (hidden) {
+            return <span key={idx} style={{display: 'none'}}>{Component}</span>;
```
Comment (alansouzati) [inline src/js/components/Split.js]: any reason why we are using a `span` instead of a `div` ?

Comment (linde12) [inline src/js/components/Split.js]: @alansouzati No, no specific reason. Want me to change it to a div?

Comment (alansouzati) [inline src/js/components/Split.js]: yeap, i believe it would be better

**Pull request merged**